project2/py/web_scraper.py

```diff
@@ -45,6 +46,42 @@ def parse_ingredients(ingredients):
     objs = utils.pmap(Ingredient, ingredients)
     return objs
 
+def parse_methods(directions, kb):
+    allMethods = []
+    for direction in directions:
+        direction = direction.lower().replace('.,', '')
+        allMethods.extend(extract_methods(direction, kb))
+
+    allMethods = list(set(allMethods))
+    return allMethods
+
+def parse_tools(directions, kb):
+    allTools = []
+    for direction in directions:
+        direction = direction.lower().replace('.,', '')
+        allTools.extend(extract_tools(direction, kb))
+
+    allTools = list(set(allTools))
+    return allTools
+
+def extract_methods(direction, kb):
+    methods = []
+    for word in kb.methods:
+        if word in direction:
+            methods.append(word.capitalize())
+    return methods
+
+def extract_tools(direction, kb):
```

Comment:
Is the only difference between `extract_methods` and `extract_tools` the `kb` collection? seems like we could just pass in the collection instead of the kb and have one function instead of two. 

That being said this will work as is and if you don't have time to change it that's totally fine, go ahead and merge away!

Reply:
good suggestion!